core/src/main/java/com/linecorp/armeria/server/file/AbstractHttpFileBuilder.java

```diff
@@ -199,7 +204,46 @@ public final B setHeader(CharSequence name, Object value) {
      */
     public final B setHeaders(Headers<AsciiString, String, ?> headers) {
         requireNonNull(headers, "headers");
-        getOrCreateHeaders().setAll(headers);
+        if (!headers.isEmpty()) {
+            getOrCreateHeaders().setAll(headers);
+        }
         return self();
     }
+
+    /**
+     * Sets the {@code "content-type"} header. This method is a shortcut of:
+     * <pre>{@code
+     * builder.autoDetectedContentType(false);
+     * builder.setHeader(HttpHeaderNames.CONTENT_TYPE, contentType);
+     * }</pre>
+     */
+    public final B contentType(MediaType contentType) {
+        requireNonNull(contentType, "contentType");
+        autoDetectedContentType(false);
+        return setHeader(HttpHeaderNames.CONTENT_TYPE, contentType);
```

Comment:
Just curious. What if we use `getOrCreateHeaders().contentType(contentType)`?

Reply:
No problem. :-)